### PR TITLE
Spawn API

### PR DESCRIPTION
**Pull request by chriso**
This partially brings back #82, giving guests the ability to spawn and kill processes.

### CODE REVIEW

File: internal/timecraft/process.go [hunk 1]
```diff
@@ -156,6 +162,26 @@ func (pm *ProcessManager) Start(moduleSpec ModuleSpec, logSpec *LogSpec) (Proces
 		}
 	}()
 
+	infc, err := netns.InterfaceByName("en0")
```
Comment (chriso) [inline internal/timecraft/process.go:165]: Should we avoid hard-coding the interface name here? Maybe instead we could loop over interfaces, skip the loopback, and then return the first address we find?

File: proto/timecraft/server/v1/timecraft.proto [hunk 2]
```diff
@@ -90,17 +90,39 @@ message ProcessIDResponse {
   string process_id = 1;
 }
 
+message SpawnRequest {
+  ModuleSpec module = 1;
+}
+
+message SpawnResponse {
+  string process_id = 1;
+  string ip_address = 2;
```
Comment (chriso) [inline proto/timecraft/server/v1/timecraft.proto:99]: Not sure if we want to return both the ipv4 and ipv6 address when calling spawn, or whether just the ipv4 address is fine?

File: internal/timecraft/process.go [hunk 3]
```diff
@@ -59,10 +59,16 @@ type ProcessInfo struct {
 	// ID is the ID of the process.
 	ID ProcessID
 
+	// Addr is a unique IP address for the process.
+	Addr netip.Addr
+
```
Comment (achille-roussel) [inline internal/timecraft/process.go:64]: This is good enough for now but I think we'll want to enhance the model later to have a representation of each network interface exposed to the process :)

File: internal/timecraft/process.go (hunk 1 above)
Comment (achille-roussel) [reply internal/timecraft/process.go:165]: I'd say let's keep it as-is for now, I think we'll revisit those soon enough that it's not a big concern 👍

File: proto/timecraft/server/v1/timecraft.proto (hunk 2 above)
Comment (achille-roussel) [reply proto/timecraft/server/v1/timecraft.proto:99]: Related to my other comment, I think we'll want to have a more fleshed out description of the process configuration, including the list of network interfaces and their addresses.

**Review by achille-roussel — APPROVED** [review]

**Pull request merged**